raiden/transfer/node.py

```diff
@@ -732,156 +747,191 @@ def handle_update_transport_authdata(
         iteration = handle_token_network_action(
             chain_state,
             state_change,
         )
     elif type(state_change) == ContractReceiveRouteClosed:
+        assert isinstance(state_change, ContractReceiveRouteClosed), MYPY_ANNOTATION
         iteration = handle_token_network_action(
             chain_state,
             state_change,
         )
     elif type(state_change) == ContractReceiveSecretReveal:
-        iteration = handle_secret_reveal(
+        assert isinstance(state_change, ContractReceiveSecretReveal), MYPY_ANNOTATION
+        iteration = handle_contract_secret_reveal(
             chain_state,
             state_change,
         )
     elif type(state_change) == ContractReceiveUpdateTransfer:
+        assert isinstance(state_change, ContractReceiveUpdateTransfer), MYPY_ANNOTATION
         iteration = handle_token_network_action(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveDelivered:
+        assert isinstance(state_change, ReceiveDelivered), MYPY_ANNOTATION
         iteration = handle_delivered(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveSecretReveal:
+        assert isinstance(state_change, ReceiveSecretReveal), MYPY_ANNOTATION
         iteration = handle_secret_reveal(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveTransferRefundCancelRoute:
+        assert isinstance(state_change, ReceiveTransferRefundCancelRoute), MYPY_ANNOTATION
         iteration = handle_receive_transfer_refund_cancel_route(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveTransferRefund:
+        assert isinstance(state_change, ReceiveTransferRefund), MYPY_ANNOTATION
         iteration = handle_receive_transfer_refund(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveSecretRequest:
+        assert isinstance(state_change, ReceiveSecretRequest), MYPY_ANNOTATION
         iteration = handle_receive_secret_request(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveProcessed:
+        assert isinstance(state_change, ReceiveProcessed), MYPY_ANNOTATION
         iteration = handle_processed(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveUnlock:
+        assert isinstance(state_change, ReceiveUnlock), MYPY_ANNOTATION
         iteration = handle_receive_unlock(
             chain_state,
             state_change,
         )
     elif type(state_change) == ReceiveLockExpired:
+        assert isinstance(state_change, ReceiveLockExpired), MYPY_ANNOTATION
         iteration = handle_receive_lock_expired(
             chain_state,
             state_change,
         )
 
+    assert chain_state is not None, 'chain_state must be set'
```

Comment:
@hackaugusto There is a case where the `chain_state` will be `None` here and that is at the very beginning when we process the `ActionInitChain` state change if there is no snapshot.